### PR TITLE
[WebAssembly] Simplify a switch-case in CFGStackify (NFC)

### PR DESCRIPTION
**Pull request by aheejin**
This merges some `case`s using `[[fallthrough]]`, and make `DELEGATE` as a separate `case`. (Previously the reason we didn't do that was not to duplicate the code in `RewriteOperands`. But now that we've extracted it into a lambda function in #107182 we can do it.

### CODE REVIEW

File: llvm/lib/Target/WebAssembly/WebAssemblyInstrControl.td [hunk 1]
```diff
@@ -147,9 +147,9 @@ let isTerminator = 1, hasSideEffects = 1, isBarrier = 1, hasCtrlDep = 1,
 // usage gets low enough.
 
 // Rethrowing an exception: rethrow
-let isTerminator = 1, hasCtrlDep = 1, isBarrier = 1 in {
+let isTerminator = 1, hasCtrlDep = 1, isBarrier = 1 in
 defm RETHROW : NRI<(outs), (ins i32imm:$depth), [], "rethrow \t$depth", 0x09>;
-} // isTerminator = 1, hasCtrlDep = 1, isBarrier = 1
+// isTerminator = 1, hasCtrlDep = 1, isBarrier = 1
```
Comment (aheejin) [inline llvm/lib/Target/WebAssembly/WebAssemblyInstrControl.td:152]: Drive-by fix

**Review by dschuff — APPROVED** [review]
